datahub/bed_api/models.py

```diff
@@ -0,0 +1,249 @@
+import copy
+
+from datahub.bed_api.constants import (
+    BusinessArea,
+    Classification,
+    ContactType,
+    DepartmentEyes,
+    HighLevelSector,
+    InteractionType,
+    IssueType,
+    JobType,
+    LowLevelSector,
+    RegionOrInternational,
+    Salutation,
+    Sentiment,
+    TopIssuesByRank,
+    TransparencyStatus,
+)
+from datahub.bed_api.utils import remove_blank_from_dict
+
+
+class BedEntity:
+    """Represents a BED  base model"""
+
+    def __init__(self):
+        """Constructor for Id based BED entity"""
+        self.Id = None
+
+    def as_values_only_dict(self):
+        """
+        Generates dictionary with no empty values to optimise data being passed as a dictionary
+        Typical scenario is for new records being posted.
+        NOTE: If updating be sure to include all values, even blanks,
+                as blank may be valid on an update.
+        """
+        result = remove_blank_from_dict(self.as_all_values_dict())
+        return result
+
+    def as_all_values_dict(self):
+        """
+        Utilises the internal dictionary to generate all values even if blank
+        :return: Generated dictionary of all class values as name value pair
+        """
+        result = copy.deepcopy(self.__dict__)
+        return result
```

Comment:
```suggestion
        return copy.deepcopy(self.__dict__)
```

Similar comment to above (:+1: on the `copy.deepcopy` though)